src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1071,38 +1034,162 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
+            numberOfArcherBarbarians = 16;
+            numberOfChiefBarbarians = 2;
+        }
+
+            numberOfBarbarians = hordeTotal - maxSize;
+            if (numberOfBarbarians < 0) {numberOfBarbarians = 0;} //For error handling and correct hordeTotal
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            if (hordeTotal > maxSize)
+            {
+                numberOfArcherBarbarians = hordeTotal - maxSize;
+                if (numberOfArcherBarbarians < 0) {numberOfArcherBarbarians = 0;} //For error handling and correct hordeTotal
+                hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+                if (hordeTotal > maxSize)
+                {
+                    numberOfChiefBarbarians = hordeTotal - maxSize;
+                    if (numberOfChiefBarbarians < 0) { numberOfChiefBarbarians = 0; //For error handling's sake, shouldn't ever happen, but would prefer not to crash if it does.
+                }
+                }
+            }
+        }
+
+        int x = this.getCenter().getX();
+        int y = this.getCenter().getY();
+        int z = this.getCenter().getZ();
+
+        assert raidingWorld != null; //Make sure world isn't null
+        switch(raidingWorld.rand.nextInt(7))
+        {
+            case 0:
+                x += Configurations.workingRangeTownHall+20;
+                break;
+            case 1:
+                x -= Configurations.workingRangeTownHall+20;
+                break;
+            case 2:
+                z += Configurations.workingRangeTownHall+20;
+                break;
+            case 3:
+                z -= Configurations.workingRangeTownHall+20;
+                break;
+            case 4:
+                x += Configurations.workingRangeTownHall+20;
+                z += Configurations.workingRangeTownHall+20;
+                break;
+            case 5:
+                x += Configurations.workingRangeTownHall+20;
+                z -= Configurations.workingRangeTownHall+20;
+                break;
+            case 6:
+                x -= Configurations.workingRangeTownHall+20;
+                z += Configurations.workingRangeTownHall+20;
+                break;
+            case 7:
+                x -= Configurations.workingRangeTownHall+20;
+                z -= Configurations.workingRangeTownHall+20;
+                break;
+            default:
+                x += Configurations.workingRangeTownHall+20;
+                break;
+        }
+
+        y = raidingWorld.getTopSolidOrLiquidBlock(new BlockPos.MutableBlockPos(x,y,z)).getY(); //Make sure mob spawns on surface.
+
+        spawn(barbarian, numberOfBarbarians,x,y,z);
+        spawn(archer, numberOfArcherBarbarians,x,y,z);
+        spawn(chief, numberOfChiefBarbarians,x,y,z);
+    }
+
+    private void spawn(ResourceLocation entityToSpawn, int numberOfSpawns,int x, int y, int z) {
```

Comment:
I also think we don't need those methods in colony.
Pull them to "BabarianHelper" or "BabarianUtils"